src/diart/models.py

```diff
@@ -83,21 +98,6 @@ def sample_rate(self) -> int:
     def duration(self) -> float:
         pass
 
-    @abstractmethod
-    def forward(self, waveform: torch.Tensor) -> torch.Tensor:
-        """
-        Forward pass of the segmentation model.
-
-        Parameters
-        ----------
-        waveform: torch.Tensor, shape (batch, channels, samples)
-
-        Returns
-        -------
-        speaker_segmentation: torch.Tensor, shape (batch, frames, speakers)
-        """
```

Comment:
Let's put this docstring in the new `__call__` method

Reply:
What about the weights parameter?

Reply:
would you like me to move __call__ to `SegmentationModel`? Currently it's at `PyannoteSegmentationModel`